src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -550,22 +528,385 @@ end = struct
+        let%bind fake_network =
+          Fake_network.Generator.(
+            gen ~max_frontier_length [fresh_peer; fresh_peer])
+        in
+        let%map make_branch =
+          Transition_frontier.Breadcrumb.For_tests.gen_seq
+            ~accounts_with_secret_keys:Genesis_ledger.accounts branch_size
+        in
+        let [me; _] = fake_network.peer_networks in
+        let branch =
+          Async.Thread_safe.block_on_async_exn (fun () ->
+              make_branch (Transition_frontier.root me.state.frontier) )
+        in
+        (fake_network, branch))
+        ~f:(fun (fake_network, branch) ->
+          let [me; _] = fake_network.peer_networks in
+          let genesis_root =
+            Transition_frontier.(
+              Breadcrumb.validated_transition @@ root me.state.frontier)
+          in
+          let transition_graph = Transition_cache.create () in
+          let sync_ledger_reader, sync_ledger_writer =
+            Pipe_lib.Strict_pipe.create ~name:"sync_ledger_reader" Synchronous
+          in
+          let bootstrap =
+            make_non_running_bootstrap ~genesis_root ~network:me.network
+          in
+          let root_sync_ledger =
+            Sync_ledger.Db.create
+              (Transition_frontier.root_snarked_ledger me.state.frontier)
+              ~logger ~trust_system
+          in
+          let start_time = Block_time.now time_controller in
+          Async.Thread_safe.block_on_async_exn (fun () ->
+              let sync_deferred =
+                sync_ledger bootstrap ~root_sync_ledger ~transition_graph
+                  ~sync_ledger_reader
+              in
+              let%bind () =
+                Deferred.List.iter branch ~f:(fun breadcrumb ->
+                    Strict_pipe.Writer.write sync_ledger_writer
+                      ( `Transition (downcast_breadcrumb breadcrumb)
+                      , `Time_received (Block_time.add start_time one_second)
+                      ) )
+              in
+              Strict_pipe.Writer.close sync_ledger_writer ;
+              sync_deferred ) ;
+          let expected_transitions =
+            List.map branch ~f:(fun breadcrumb ->
+                Transition_frontier.Breadcrumb.validated_transition breadcrumb
+                |> External_transition.Validation.forget_validation )
+          in
+          let saved_transitions =
+            Transition_cache.data transition_graph
+            |> List.map ~f:(fun env ->
+                   let transition, _ = Envelope.Incoming.data env in
+                   transition.data )
+          in
+          [%test_result: External_transition.Set.t]
+            (External_transition.Set.of_list saved_transitions)
+            ~expect:(External_transition.Set.of_list expected_transitions) )
+
+    (*
+    let run_bootstrap ~timeout_duration ~my_net ~transition_reader ~should_ask_best_tip =
```

Comment:
Can we have these tests back on before you merge your PR?

Reply:
Yep, that's the plan. @bkase asked me to get this landed ASAP, but there will be followup issues to reintegrate these tests soon. We decided we are ok with temporarily disabling these unit tests since we aren't actively changing these files atm, but we need to get them back in within the next week or two.